src/pages/workspace/WorkspaceJoinUserPage.tsx

```diff
@@ -30,7 +29,8 @@ function WorkspaceJoinUserPage({route, policies}: WorkspaceJoinUserPageProps) {
     const styles = useThemeStyles();
     const policyID = route?.params?.policyID;
     const inviterEmail = route?.params?.email;
-    const policy = ReportUtils.getPolicy(policyID);
+    const policy = useMemo(() => policies?.[`${ONYXKEYS.COLLECTION.POLICY}${policyID}`] ?? ({} as Policy), [policies, policyID]);
```

Comment:
My first instinct was that this component doesn't need to connect to the entire policy collection. It looks like the only reason it connects to the entire collection is pass all policies to `PolicyUtils.isPolicyMember()`. All that method does is check that Onyx contains a policy with that ID (which doesn't necessarily mean the user is a member).

If that's all that `isPolicyMember()` is looking for, then it seems like `isPolicyMember` would be `true` as long as the `policy` object returned here is not an empty object.

Following that logic, then the full policy collection doesn't need to be subscribed to and `isPolicyMember` can be replaced with `!isEmptyObject(policy)`.

Reply:
@tgolen Thanks your suggestions. I just updated code, please check again @DylanDylann